File: backend/api/routes/check.py
# This file is part of the Official Document AI Assistant.
# (c) 2026 Jose AI (https://www.linhut.cn)
# Licensed under the MIT License. See the LICENSE file for details.
"""
Check API routes: format checking and issue management.
"""
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from db.database import get_db
from api.schemas.api_models import (
    CheckRequest, CheckResultResponse, CheckIssueResponse, IssueActionRequest
)
from services import document_service as svc

router = APIRouter()


@router.post("/{doc_id}", response_model=CheckResultResponse)
async def run_check(doc_id: int, req: CheckRequest | None = None, db: Session = Depends(get_db)):
    """Run format checks on a document."""
    doc_type = req.document_type if req else None
    try:
        result = svc.check_document(db, doc_id, doc_type)
    except ValueError as e:
        raise HTTPException(status_code=404, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

    issues = svc.get_check_results(db, doc_id)
    return CheckResultResponse(
        document_id=result["document_id"],
        total_issues=result["total_issues"],
        p0_count=result["p0_count"],
        p1_count=result["p1_count"],
        p2_count=result["p2_count"],
        issues=[
            CheckIssueResponse(
                id=i.id, check_type=i.check_type, severity=i.severity,
                rule_id=i.rule_id, location=i.location,
                original_text=i.original_text, suggested_fix=i.suggested_fix,
                reason=i.reason, status=i.status,
            ) for i in issues
        ],
    )


@router.get("/{doc_id}/results", response_model=list[CheckIssueResponse])
async def get_results(doc_id: int, db: Session = Depends(get_db)):
    """Get check results for a document."""
    issues = svc.get_check_results(db, doc_id)
    return [
        CheckIssueResponse(
            id=i.id, check_type=i.check_type, severity=i.severity,
            rule_id=i.rule_id, location=i.location,
            original_text=i.original_text, suggested_fix=i.suggested_fix,
            reason=i.reason, status=i.status,
        ) for i in issues
    ]


@router.put("/{doc_id}/issues/{issue_id}")
async def update_issue(
    doc_id: int, issue_id: int,
    req: IssueActionRequest,
    db: Session = Depends(get_db),
):
    """Accept or dismiss a check issue."""
    status_map = {"accept": "accepted", "dismiss": "dismissed"}
    new_status = status_map.get(req.action)
    if not new_status:
        raise HTTPException(status_code=400, detail="action must be 'accept' or 'dismiss'")
    ok = svc.update_issue_status(db, issue_id, new_status)
    if not ok:
        raise HTTPException(status_code=404, detail="Issue not found")
    return {"status": new_status}

File: frontend/src/api/check.ts
/*
 * This file is part of the Official Document AI Assistant.
 * (c) 2026 Jose AI (https://www.linhut.cn)
 * Licensed under the MIT License. See the LICENSE file for details.
 */
/**
 * Check API - 检查相关接口
 *
 * 路径与后端 backend/api/routes/check.py 一致：
 *   POST /api/check/{doc_id}           — 执行格式检查
 *   GET  /api/check/{doc_id}/results   — 获取检查结果
 *   PUT  /api/check/{doc_id}/issues/{issue_id} — 应用/忽略修复
 */
import apiClient from './client';

export interface CheckResult {
  id: number;
  document_id: number;
  check_type: string;
  severity: 'P0' | 'P1' | 'P2';
  rule_id?: string;
  location?: string;
  original_text?: string;
  suggested_fix?: string;
  reason?: string;
  status: 'pending' | 'accepted' | 'dismissed';
}

/**
 * 执行格式检查
 */
export const runCheck = async (docId: number, documentType?: string): Promise<unknown> => {
  return apiClient.post(`/api/check/${docId}`, { document_type: documentType });
};

/**
 * 获取检查结果
 */
export const getCheckResults = async (docId: number): Promise<CheckResult[]> => {
  return apiClient.get(`/api/check/${docId}/results`);
};

/**
 * 应用修复建议
 */
export const applyFix = async (docId: number, issueId: number): Promise<void> => {
  return apiClient.put(`/api/check/${docId}/issues/${issueId}`, { action: 'accept' });
};

/**
 * 忽略问题
 */
export const dismissIssue = async (docId: number, issueId: number): Promise<void> => {
  return apiClient.put(`/api/check/${docId}/issues/${issueId}`, { action: 'dismiss' });
};
